src/runner/test/assert.js

```diff
@@ -210,11 +210,10 @@ module.exports = function createAssert(scope, testCode, settings){
 
         try {
           asyncQueue.shift().call();
+          __processAsync();
```

Comment:
On exception test gets done in `__exception` call . If call `__processAsync()` after `try/catch` clause `testDone` is called twice.